### ISSUE TITLE
Hello message

### ISSUE THREAD

**Issue by cpacejo** [org-zpr/zpr-core#265]
To test the request/reply functionality, let's implement the Hello request and reply.  These are specified in [§ 6.3.4](https://docs.google.com/document/d/1W0NDiHSKVvnVx5QytaxD00uUaoA6t9YU/edit#heading=h.42ddq1a).  Notably, the `Hello` message is empty (so we don't need to define a struct for it).  Define a struct for the response, call it `ZdpHelloResponseHeader`, but put only one field in it, `status`, a `u16`.  (Ignore the other fields for now… we'll circle back to them once SAs and link state is in place.)

In the inbound processor, you'll now need to handle messages of type `HelloRequest` – for these, you'll just want to send a message (using `enqueue_non_flow_pkt()`) of type `HelloReply` with a `ZdpHelloResponseHeader` body (hardcode `status` as 0 for now).  You can even reuse the same packet buffer like `Packet::new(pkt.destroy(), ...)`.  We can also print a message to stderr saying we got the `HelloRequest`.

Now, at startup, create a new task, which calls into the sync-request framework you made to send a `Hello` message.  It will need to get a buffer from the buffer stack, make it a packet, and "fill it in", but since the `Hello` message has no body there's nothing to do here.  Then it can just wait for a response, decode it as a `ZdpHelloResponseHeader`, and for now just print a message which also includes the value of the status field.

Now when you start the system, whichever half starts later should successfully send and receive a Hello message!  (The other half won't, because it won't retry.)  This is fine for now as it's just a test – later when we get link establishment going we'll have the `Hello` exchange wait for the SA to be established.

Blocked on:
- org-zpr/zpr-core#264
